src/scripts/main.js

```diff
@@ -1,3 +1,334 @@
-'use strict';
+'use strict'
 
-// write your code here
+const BUTTON_CLASS = 'button';
+const FIELD_ROW_CLASS = 'field-row';
+const FIELD_CELL_CLASS = 'field-cell';
+const GAME_SCORE_CLASS = 'game-score';
+const START_MESSAGE_CLASS = 'message-start';
+const LOSE_MESSAGE_CLASS = 'message-lose';
+const WIN_MESSAGE_CLASS = 'message-win';
+const HIDDEN_CLASS = 'hidden';
+
+const RESTART_TEXT = 'Restart';
+const START_TEXT = 'Start';
+
+const FIELD_SIZE = 4;
+const WIN_SCORE = 2048;
+
+class BaseClass {
+  constructor() {
+    this.score = 0;
+    this.fields = [];
+    this.isGameOver = false;
+    this.isGameStart = false;
+    this.restartClass = RESTART_TEXT.toLocaleLowerCase();
+    this.startClass = START_TEXT.toLocaleLowerCase();
+
+    this.setSelectors();
+  }
+
+  setSelectors() {
+    this.buttonEl = document.getElementsByClassName(BUTTON_CLASS)[0];
+    this.fieldRowsEl = document.getElementsByClassName(FIELD_ROW_CLASS);
+    this.gameScoreEl = document.getElementsByClassName(GAME_SCORE_CLASS)[0];
+    this.loseMessageEl = document.getElementsByClassName(LOSE_MESSAGE_CLASS)[0];
+    this.winMessageEl = document.getElementsByClassName(WIN_MESSAGE_CLASS)[0];
```

Comment:
sugggest to use docuement.querySelector method. it will return element instead colletction and will be no need to take first element from colletction.
for start button better to find by .start class.

Reply:
fixed